clustering/gaia_wrapper.py

```diff
@@ -0,0 +1,113 @@
+import logging
+import os
+import sys
+import time
+import yaml
+from django.conf import settings
+
+from gaia2 import DataSet, View, DistanceFunctionFactory
+
+import clustering_settings as clust_settings
+
+
+logger = logging.getLogger('clustering')
+
+
+class GaiaWrapperClustering:
+    """Gaia wrapper for the clustering engine.
+
+    This class contains helper methods to interface with Gaia.
+    When creating the instance object, Gaia datasets corresponding to the features configured in the clustering 
+    settings file will be loaded.
+    """
+    def __init__(self):
+        self.index_path = clust_settings.INDEX_DIR
+        self.__load_datasets()
+
+    def __get_dataset_path(self, ds_name):
+        return os.path.join(clust_settings.INDEX_DIR, ds_name + '.db')
+
+    def __load_dataset(self, features, gaia_index, gaia_descriptor_names, gaia_metric):
+        """Loads a Gaia dataset, view and metric for a specific feature and config.
+        
+        Args: 
+            features (str): name of the features to be used for clustering. 
+                Available features are listed in the clustering settings file
+            gaia_index (str): Name of the Gaia dataset index file.
+            gaia_descriptor_names (str or List[str]): Name(s) of the descriptor field to use within the Gaia dataset. 
+            gaia_metric (str): Name of the metric to use for nearest neighbors search.
+        """
+        gaia_dataset = DataSet()
+        gaia_dataset.load(self.__get_dataset_path(gaia_index))
+        gaia_view = View(gaia_dataset)
+        gaia_metric = DistanceFunctionFactory.create(gaia_metric, gaia_dataset.layout(), 
+            {'descriptorNames': gaia_descriptor_names})
+
+        setattr(self, '{}_dataset'.format(features), gaia_dataset)
+        setattr(self, '{}_view'.format(features), gaia_view)
+        setattr(self, '{}_metric'.format(features), gaia_metric)
```

Comment:
I agree with this comment - I think that frederic's solution is more understandable instead of having to always use getattr/setattr whenever you want to get a feature. It makes the class a bit more self-documenting as well, as you can define this field in the class definition and don't need to read the code to work out that there are additional fields floating around